lib/csv/parser.rb

```diff
@@ -575,9 +575,9 @@ def build_scanner
           string = @samples[0]
         end
         if string
-          unless string.valid_encoding?
+          if index = string.lines.index { |l| !l.valid_encoding? }
```

Comment:
We need to respect `@row_separator`: `string.lines(@raw_separator)`.

Reply:
I assume that `@raw_separator` was just a typo for `@row_separator`?

Reply:
Ah, sorry. You're right.